metricbeat/module/http/json/json.go

```diff
@@ -150,6 +135,48 @@ func (m *MetricSet) Fetch() (common.MapStr, error) {
 	// Set dynamic namespace
 	event["_namespace"] = m.namespace
 
+	return event
+}
+
+// Fetch methods implements the data gathering and data conversion to the right format
+// It returns the event which is then forward to the output. In case of an error, a
+// descriptive error must be returned.
+func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+	response, err := m.http.FetchResponse()
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+
+	var jsonBody interface{}
+	var event []common.MapStr
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(body, &jsonBody)
+	if err != nil {
+		return nil, err
+	}
+
+	switch body := jsonBody.(type) {
```

Comment:
Let's separate the processing and not only do a "dynamic" switch to figure out if there is an error. The way I imagine is that we have jsonBody and jsonBodyArray (or similar) and directly decode in what we want in the end. Like this I would assume we don't have to do any type checks and directly get the errors.